### PR TITLE
Feature/10 dse has a slack home 2

### LINKED ISSUE #10: DSE has a Slack home screen with fake data

### PR DESCRIPTION
**Pull request by Startouf**
Implements partially #10 

Adds a section that is dynamically filled from the JSON example.

next step is to start handling message actions but I really need Smaril's #17 for this

### CODE REVIEW

File: apphome/messages_awaiting_enlightenment.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	EvilIndex     string `json:"evilIndex"`
+	Date          string `json:"date"`
+	Channel       string `json:"channel"`
+	MessageLink   string `json:"messageLink"`
+	Original      string `json:"original"`
+	DSESuggestion string `json:"dseSuggestion"`
+}
+
+func EnlightenmentSection() []slack.Block {
+	// Hello
```
Comment (Startouf) [inline apphome/messages_awaiting_enlightenment.go]:
```suggestion
```

File: apphome/apphome.go [hunk 2]
```diff
@@ -3,35 +3,78 @@ package apphome
...
+		heredoc.Doc(fmt.Sprintf(`
+			*Top Channels with evil messages*
+			:airplane: General · 30%% (142)
+			:taxi: Code Reviews · 66%% (43)
+			:knife_fork_plate: Direct Messages · 18%% (75)`,
+		)),
+		false, false,
+	)
+	fields := []*slack.TextBlockObject{messageStats, topChannelsText}
+	//
```
Comment (Startouf) [inline apphome/apphome.go]:
```suggestion
```

File: apphome/apphome.go [hunk 3]
```diff
@@ -3,35 +3,78 @@ package apphome
...
 
-	return message
+	return []slack.Block{headerSection, helloSection, slack.NewDividerBlock()}
 }
 
-func translateUserIdToUserName(userId string) string {
-	// TODO in https://github.com/gjgd/dont-slack-evil/issues/16
-	return userId
+func statsSections() []slack.Block {
+	// MessageStats
```
Comment (Startouf) [inline apphome/apphome.go]:
```suggestion
```

File: apphome/apphome.go [hunk 4]
```diff
@@ -3,35 +3,78 @@ package apphome
...
+}
+
+func introSections(userName string) []slack.Block {
 	// Header Section
 	headerOptionsTxt := slack.NewTextBlockObject("plain_text", "Manage App Settings", true, false)
 	headerButton := slack.NewButtonBlockElement("", "app_settings", headerOptionsTxt)
 	headerText := slack.NewTextBlockObject("mrkdwn", "*Don't Slack Evil Performance*", false, false)
 	headerSection := slack.NewSectionBlock(headerText, nil, slack.NewAccessory(headerButton))
 
 	// Hello
```
Comment (Startouf) [inline apphome/apphome.go]:
```suggestion
```

File: apphome/apphome.go [hunk 5]
```diff
@@ -3,35 +3,78 @@ package apphome
...
+	)
+}
+
+func translateUserIdToUserName(userId string) string {
+	// TODO in https://github.com/gjgd/dont-slack-evil/issues/16
+	return userId
+}
+
+func introSections(userName string) []slack.Block {
 	// Header Section
```
Comment (Startouf) [inline apphome/apphome.go]:
```suggestion
```

File: apphome/messages_awaiting_enlightenment.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+	for _, message := range messages {
+		blocks = append(blocks, formatMessageToBeEnlightened(message)...)
+	}
+	return blocks
+}
+
+func parseTestMessages() []Message {
+	jsonFile, err := os.Open("sample.json")
+	// if we os.Open returns an error then handle it
```
Comment (Startouf) [inline apphome/messages_awaiting_enlightenment.go]:
```suggestion
```

File: apphome/messages_awaiting_enlightenment.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer jsonFile.Close()
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	// we initialize our Users array
```
Comment (Startouf) [inline apphome/messages_awaiting_enlightenment.go]:
```suggestion
```

File: helpers/helpers.go [hunk 8]
```diff
@@ -15,3 +19,8 @@ func ReverseRunes(s string) string {
 	}
 	return string(r)
 }
+
+func QuoteForSlack(message string) string {
+	var re = regexp.MustCompile(`(.*)`)
+	return re.ReplaceAllString(message, `> $1`)
```
Comment (Startouf) [inline helpers/helpers.go:25]: Hey @mourtisma j'ai pas réussi à trouver comment faire une expression régulière qui ne remplace pas les lignes vides par des `> `, tu as des billes ?

Genre 

```
Hello
How are you ?

I'm fina thanks !
```
-->
```
> Hello
> How are you ?

> I'm fina thanks !
```

File: helpers/helpers.go [hunk 9]
```diff
@@ -15,3 +19,8 @@ func ReverseRunes(s string) string {
 	}
 	return string(r)
 }
+
+func QuoteForSlack(message string) string {
+	var re = regexp.MustCompile(`(.*)`)
```
Comment (gjgd) [inline helpers/helpers.go]:
```suggestion
	var re = regexp.MustCompile(`(.+)`)
```

File: helpers/helpers.go (hunk 8 above)
Comment (gjgd) [reply helpers/helpers.go:25]: Try `(.+)` to avoid blank lines

**Review by gjgd — APPROVED** [review]

File: helpers/helpers.go (hunk 8 above)
Comment (Startouf) [reply helpers/helpers.go:25]: :facepalm: I had tried other regex like`^[^ ].*$)` but the `^` and `$` behaved strangely in go. Hopefully you had a more straightforward regex I should have thought of, thx !

**Pull request merged**